### DISCUSSION TITLE
Problem w/ CLI dump matchups

### DISCUSSION THREAD

**Discussion by Ace7866** [mattdodge/yahoofantasy#22] [Q&A] [answered]
Hi guys - I'm noticing an issue with the CLI dump command for the matchups. The command seems unable to change the season and all of my dumps are pulling data from 2018 regardless of which season I designate with the '-s 2016' command. I've also tried with the more general 'yahoofantasy dump matchups' and then designating a different season that way. 

No matter which method I try, or what year/season I designate, I get the same results each time - the matchups from the 2018 season. 

Could anyone else try this to see if they can replicate the issue? Wondering if I'm doing something wrong? Or maybe a problem with my league itself? 

Any tips on how to troubleshoot would be much appreciated!

**Comment by mattdodge** [comment, marked as answer]
I can't think of why it would be defaulting to 2018, that seems weird. The CLI is a bit finicky in that you have to specify the season after the `dump` command but before the type (`matchups`), maybe that's part of the issue? Entering it when prompted should fix that though.

That being said, I just dumped my 2016 MLB matchups no problem with this command `yahoofantasy dump -s 2016 matchups`. What game (MLB?) are you trying to dump matchups for?

There's not really a debug mode on the CLI unfortunately, but you can try firing up the league context manually and seeing what leagues it is returning:
```python
from yahoofantasy import Context
ctx = Context()
leagues = ctx.get_leagues('mlb', 2016)
for league in leagues:
  print(league.id, league.name)
```
Make sure those league names and IDs look correct.

**Reply by Ace7866** [reply]
![image](https://user-images.githubusercontent.com/58792096/135951122-0e52fb9a-0ec5-41a2-b4a6-30ae42fff911.png)

![image](https://user-images.githubusercontent.com/58792096/135951176-3b0a5ddd-8665-481d-820c-e9009e2c0e72.png)

First, thank you for fast response! I tried to recreate the issue this evening and noticed it behaved similarly. Rather than being stuck on 2018, it seems that it's stuck on whatever year I pull first. So in this case I used the CLI dump to pull 2016 NFL matchup data for our league and it worked! Woot! Then I repeated the CLI dump except for 2017. Curiously, and similar to yesterday, this dump gave me all of the same exact data from the initial dump. In other words, even though I instructed it to dump 2017 data, it just gave me the 2016 data from my initial pull. Thoughts? I included some screenshots so you don't think I'm crazy.

**Reply by Ace7866** [reply]
I will just try to disconnect the Yahoo API/link and do each pull separately and independently rather than successively. Seems like that might get the job done.

**Reply by mattdodge** [reply]
Ah yes, that does sound like a bug then. It probably has to do with the resource caching, it's not properly segmenting by year I bet. I'll look into this bug to fix it. In the meantime, you can clear your cache by removing the .yahoofantasy file that gets created after you run one of your exports.

**Reply by Ace7866** [reply]
Ahh, makes sense. Wonderful. I'll go ahead and do that.
